packages/notifi-web-push-service-worker/lib/notifi-service-worker.js

```diff
@@ -0,0 +1,201 @@
+
+    const targetGroups = await client.getTargetGroups();
+    const defaultTargetGroup = targetGroups.find((targetGroup) => targetGroup.name === 'Default');
+    const webPushTargetIds = defaultTargetGroup?.webPushTargets?.map((t) => t?.Id) ?? [];
+
+    const webPushTargetResponse = await client.createWebPushTarget({
+      vapidPublicKey: vapidPublicKey,
+      endpoint: subscriptionJson.endpoint,
+      auth: subscriptionJson.keys.auth,
+      p256dh: subscriptionJson.keys.p256dh
+    })
+
+    if (!webPushTargetResponse.createWebPushTarget.webPushTarget) {
+      console.error('Failed to create web push target. CreateWebPushTargetMutation failed.');
+      return;
+    }
+
+    webPushTargetIds.push(webPushTargetResponse.createWebPushTarget.webPushTarget?.Id);
+
+    await client.ensureTargetGroup({
+      name: 'Default',
+      emailAddress: defaultTargetGroup?.emailTargets[0]?.emailAddress,
+      phoneNumber: defaultTargetGroup?.smsTargets[0]?.phoneNumber,
+      telegramId: defaultTargetGroup?.telegramTargets[0]?.telegramId,
+      discordId: defaultTargetGroup?.discordTargets[0]?.name,
+      slackId: defaultTargetGroup?.slackChannelTargets[0]?.name,
+      walletId: defaultTargetGroup?.web3Targets[0]?.name,
+      webPushTargetIds: webPushTargetIds
+    });
+  }
+  catch (err) {
+    console.error(err, "Failed to create web push target.")
+  }
+}
+
+function GetSubsciption(userAccount, dappId, env) {
+  if (Notification.permission !== "granted") {
+    console.log('Notification permissions not granted');
+    return;
+  }
+
+  if (!userAccount || !dappId || !env || userAccount == '' || dappId == '' || env == '') {
+    console.log('UserAccount, Notifi dappId, or env not found. Skipping subscription instantiation.');
+    return;
+  }
+
+  // TODO: Instantiate Notifi client here. If it fails, don't do anything.
+  client = instantiateFrontendClient(
+    dappId,
+    {
+      walletBlockchain: 'OFF_CHAIN',
+      userAccount: userAccount
+    },
+    env,
+    undefined,
+    { fetch }
+  );
+
+  client.initialize().then(userState => {
+    if (userState.status == 'authenticated') {
+      // TODO: Get vapid key here
+      let vapidPublicKey = "BBw1aI15zN4HFMIlbWoV2E390hxgY47-mBjN41Ewr2YCNGPdoR3-Q1vI-LAyfut8rqwSOWrcBA5sA5aC4gHcFjA";
+
+      if (Notification.permission === "granted") {
+        self.registration.pushManager.getSubscription()
+          .then(async (subscription) => {
+            if (subscription) {
+              console.log('subscription already exists')
+              return subscription;
+            }
+
+            const convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
+
+            return await self.registration.pushManager.subscribe({
+              userVisibleOnly: true,
+              applicationServerKey: convertedVapidKey,
+            });
+          })
+          .then(async (subscription) => {
+            await createWebPushTarget(subscription, vapidPublicKey)
```

Comment:
wait, this is vapidPublicKey is used by the server to encrypt the message intended for this individual, right?

Where is the private key? The private key should be maintained/stored only here. Never leave the client.

The endpoint you need to expose above, means it's just a local endpoint that grabs the public key that's meant for this user, right?